RetailCoder.VBE/Refactorings/Rename/RenameRefactoring.cs

```diff
@@ -38,358 +40,496 @@ public void Refactor()
+
+        //Same reasoning for IsControl above
+        private static bool IsEvent(RenameModel model)
         {
-            return _model.Declarations.FirstOrDefault(item => item.Scope == declaration.Scope &&
-                              item.IdentifierName == declaration.IdentifierName &&
-                              item.DeclarationType == declarationType);
+            var declaration = GetDeclarationOfType(model, DeclarationType.Event, true);
+            return (declaration != null);
         }
 
-        private void RenameModule()
+        private static Declaration GetDeclarationOfType(RenameModel model, DeclarationType goalType, bool identifierIsOnRHS)
         {
+            var declarationName = model.Target.IdentifierName;
             try
             {
-                var component = _model.Target.QualifiedName.QualifiedModuleName.Component;
-                var module = component.CodeModule;
+                if (model.Target.IdentifierName.Contains("_"))
                 {
-                    if (module.IsWrappingNullReference)
-                    {
-                        return;
-                    }
-
-                    if (component.Type == ComponentType.Document)
-                    {
-                        var properties = component.Properties;
-                        var property = properties["_CodeName"];
-                        {
-                            property.Value = _model.NewName;
-                        }
-                    }
-                    else if (component.Type == ComponentType.UserForm)
+                    if (identifierIsOnRHS)
                     {
-                        var properties = component.Properties;
-                        var property = properties["Caption"];
-                        {
-                            if ((string)property.Value == _model.Target.IdentifierName)
-                            {
-                                property.Value = _model.NewName;
-                            }
-                            component.Name = _model.NewName;
-                        }
+                        //Events
+                        declarationName = model.Target.IdentifierName.Remove(0, model.Target.IdentifierName.IndexOf("_") + 1);
                     }
                     else
                     {
-                        module.Name = _model.NewName;
+                        //Controls
+                        declarationName = model.Target.IdentifierName.Remove(model.Target.IdentifierName.IndexOf("_"));
                     }
                 }
+                var declaration = model.Declarations.Single(item =>
+                    item.DeclarationType == goalType
+                    && (declarationName.Equals(item.IdentifierName)));
+                return declaration;
             }
-            catch (COMException)
+            catch (Exception)
```

Comment:
I replaced Exception with two catch blocks which were the only two types of catches used in the original code.  Even so, the result is the same...I pop up the same error message.  Since I do not raise any events within the new code, I'm curious as to why specific catches are preferred to catching 'everything' (expected and unexpected) - and giving the user a message, rather than being silent about the exceptions I did not expect.  Thanks.